File: src/main/java/com/domain/DbAccount.java
package com.domain;

import org.hibernate.annotations.DynamicUpdate;

import javax.persistence.*;
import java.util.Date;

/**
 * 账号类
 *
 * @author Five
 * @date create in 下午2:29 2017/11/30
 */

@Entity
@Table(name = "account")
public class DbAccount {
    @Id
    @GeneratedValue
    private Integer id;

    @Column(updatable = false)
    private String uid;

    @Column(length = 64, nullable = true, updatable = false)
    private String account;

    @Column(length = 64, nullable = true)
    private String password;

    private Integer status;

    @Column(updatable = false)
    private Date createtime;

    public DbAccount() {
    }

    public Integer getId() {
        return id;
    }
    public void setId(Integer id) {
        this.id = id;
    }

    public String getUid() {
        return uid;
    }
    public void setUid(String uid) {
        this.uid = uid;
    }

    public String getAccount() {
        return account;
    }
    public void setAccount(String account) {
        this.account = account;
    }

    public String getPassword() {
        return password;
    }
    public void setPassword(String password) {
        this.password = password;
    }

    public Integer getStatus() {
        return status;
    }
    public void setStatus(Integer status) {
        this.status = status;
    }

    public Date getCreatetime() {
        return createtime;
    }
    public void setCreatetime(Date createtime) {
        this.createtime = createtime;
    }
}

File: src/main/java/com/domain/entity/DbAccount.kt
package com.domain.entity

import java.util.*
import javax.persistence.*

/**
 * class
 *
 * @author Five0314
 * @date create in 下午2:25 2017/12/14
 */
@Entity
@Table(name = "account")
class DbAccount {

    @Id
    @GeneratedValue
    var id: Int = 0 // ID

    @Column(updatable = false)
    lateinit var accountid: String // 账号编号

    @Column(updatable = false, nullable = false)
    lateinit var uid: String // 用户编号

    @Column(length = 64, nullable = true, updatable = false)
    lateinit var account: String // 账号

    @Column(length = 64)
    var password: String? = null // 密码

    var status: Int = 0 // 状态

    @Column(updatable = false)
    lateinit var createtime: Date // 创建时间

    constructor()

    constructor(accountid: String, uid: String, account: String, password: String?, status: Int, createtime: Date) {
        this.accountid = accountid
        this.uid = uid
        this.account = account
        this.password = password
        this.status = status
        this.createtime = createtime
    }
}